### ISSUE TITLE
unable to update Jira issue priority using `update_field_value` action

### ISSUE THREAD

**Issue by stfn775** [StackStorm-Exchange/jira#65]
Hi! I have been trying to update Jira issues priority using `update_field_value` but I am getting the following error:

"errors\":{\"priority\":\"Specify the Priority (id or name) in the string format\"}

I am able to update other fields using the `update_field_value` action, and I can set the priority successfully using the `create_issue` action.

Am I doing something wrong? I have looked at the action's yaml and python scripts and I cannot see anything obvious, so it might be Jira treats priority differently from other fields?

Thanks

**Comment by stfn775** [comment]
After a bit more digging I believe the issue is with the fact that the `update_field_value` action expects values as strings, while Jira needs the priority specified as an object (to differentiate between priority ID or name).

It seems to me like the only option is to create a new action: even if we can handled the priority as a different field (and marked it as optional) we would still need to update a mandatory string-field in the same action, which would be confusing.

I will be writing a PR for the new action, but if anyone can think of a better way to handle this feel free to comment! 
Thanks

**Comment by floatingstatic** [comment]
I actually see the same problem when creating an issue:

```
response text = {\"errorMessages\":[],\"errors\":{\"priority\":\"Specify the Priority (id or name) in the string format\"}
```

This was tested using `{"priority":"High"}` as extra fields with the create_issue action. I think we probably need to address this in multiple places and we can probably do so without adding a new action under #67 

I can take a pass at this today to see if we can pre-process the parameters before sending to jira if `priority` is a field that is being added or modified. I think we can add some code to modify the type into a priority object.

**Comment by floatingstatic** [comment]
Apologies, my test was off. `create_issue` works fine if we use `{"priority":{"name":"High"}}` as `extra_fields` in the action. We can probably leave that as-is. I have a fix coming for `update_field_value`...

**Comment by floatingstatic** [comment]
@stfn775 can you give #68 a try? You can test from my branch by running: 

```
st2 pack install https://github.com/floatingstatic/stackstorm-jira=priority-field
```

**Comment by stfn775** [comment]
@floatingstatic just tested, and it works beautifully. Neat solution, thank you!

**Comment by floatingstatic** [comment]
Great, I'll just have someone +1 my work and we can get this merged in for a release. Thanks!

**Comment by floatingstatic** [comment]
Merged. Fixes are in pack version 2.5.1 which should have a build shortly.
